applications/StructuralMechanicsApplication/python_scripts/set_automated_initial_variable_process.py

```diff
@@ -0,0 +1,97 @@
+# Importing the Kratos Library
+from pathlib import Path
+import KratosMultiphysics as KM
+import KratosMultiphysics.StructuralMechanicsApplication as SMA
+from KratosMultiphysics import Logger
+from KratosMultiphysics.read_csv_table_utility import ReadCsvTableUtility
+
+def Factory(settings, Model):
+    if not isinstance(settings, KM.Parameters):
+        raise Exception("Expected input shall be a Parameters object, encapsulating a json string")
+
+    default_settings = KM.Parameters(
+        """{
+            "help"                     : "This automates the application of initial strain/stress variables using csv tables",
+            "model_part_name"          : "please_specify_model_part_name",
+            "variable_name"            : "SPECIFY_VARIABLE_NAME",
+            "hole_generatrix_axis"     : [0.0,0.0,1.0],
+            "hole_generatrix_point"    : [0.0,0.0,0.0],
+            "hole_radius_offset"       : 0.0,
+            "initial_variable_table"     : {
+                        "name"             : "csv_table",
+                        "filename"         : "sample.csv",
+                        "delimiter"        : ",",
+                        "skiprows"         : 1,
+                        "first_column_id"  : 0,
+                        "second_column_id" : 1,
+                        "na_replace"       : 0.0
+                    }
+        }""")
+    process_settings = settings["Parameters"]
+    process_settings.ValidateAndAssignDefaults(default_settings)
+    computing_model_part = Model[process_settings["model_part_name"].GetString()]
+
+    default_table_id = KM.Parameters("""{
+    "table_id": 0
+    }""")
+    process_settings["initial_variable_table"].AddValue("table_id", default_table_id)
+
+    file_path = Path(process_settings["initial_variable_table"]["filename"].GetString())
+    layer_name = file_path.name.split("_")[0]
+    layer_list = [file for file in file_path.parent.iterdir() if file.name.split("_")[0] == layer_name]
+
+    if not len(layer_list):
+        ErrorMsg = "Tables of " + layer_name + " not found"
```

Comment:
```suggestion
        ErrorMsg = f"Tables of "{layer_name} not found"
```

I strongly recommend reading up on f-strings